chain-evm/src/machine.rs

```diff
@@ -0,0 +1,211 @@
+                    // If reset_storage is set, the account's balance is
+                    // set to be Default::default().
+                    let mut account =
+                        self.state
+                            .modify_account(&address, balance, nonce, code, reset_storage);
+
+                    // iterate over the apply_storage keys and values
+                    // and put them into the account.
+                    for (index, value) in apply_storage {
+                        if value == crate::state::Value::default() {
+                            // value is full of zeroes, remove it
+                            account.storage = account.storage.remove(&index);
+                        } else {
+                            account.storage = account.storage.put(index, value);
+                        }
+                    }
+
+                    if delete_empty && account.is_empty() {
+                        self.state.delete(&address);
+                    } else {
+                        self.state.insert_or_update(address, account);
+                    }
+                }
+                Apply::Delete { address } => {
+                    self.state.delete(&address);
+                }
+            }
+        }
+
+        // save the logs
+        for log in logs {
+            self.logs.push(log);
+        }
+    }
+}
+
+impl<'config> StackState<'config> for VirtualMachine {
+    fn metadata(&self) -> &evm::executor::StackSubstateMetadata<'config> {
+        todo!();
+    }
+    fn metadata_mut(&mut self) -> &mut evm::executor::StackSubstateMetadata<'config> {
+        todo!();
+    }
+    fn enter(&mut self, _gas_limit: u64, _is_static: bool) {
+        todo!();
+    }
+    fn exit_commit(&mut self) -> Result<(), evm::ExitError> {
+        todo!();
+    }
+    fn exit_revert(&mut self) -> Result<(), evm::ExitError> {
+        todo!();
+    }
+    fn exit_discard(&mut self) -> Result<(), evm::ExitError> {
+        todo!();
+    }
+    fn is_empty(&self, _address: H160) -> bool {
+        todo!();
+    }
+    fn deleted(&self, _address: H160) -> bool {
+        todo!();
+    }
+    fn inc_nonce(&mut self, _address: H160) {
+        todo!();
+    }
+    fn set_storage(&mut self, _address: H160, _key: H256, _value: H256) {
+        todo!();
+    }
+    fn reset_storage(&mut self, _address: H160) {
+        todo!();
+    }
+    fn log(&mut self, _address: H160, _topics: Vec<H256>, _data: Vec<u8>) {
+        todo!();
+    }
+    fn set_deleted(&mut self, _address: H160) {
+        todo!();
+    }
+    fn set_code(&mut self, _address: H160, _code: Vec<u8>) {
+        todo!();
+    }
+    fn transfer(&mut self, _transfer: evm::Transfer) -> Result<(), evm::ExitError> {
+        todo!();
+    }
+    fn reset_balance(&mut self, _address: H160) {
+        todo!();
+    }
+    fn touch(&mut self, _address: H160) {
+        todo!();
+    }
+}
+
+pub fn code_to_execute_evm_runtime() -> Result<(), String> {
```

Comment:
Please add a doc comment explaining this public function.
It could have a shorter name.

Reply:
Yeah, this can be ignored, it was a placeholder that never should've made it to the commit.